src/components/specific/files/files-table/file-actions-cell/FileActionsCell.vue

```diff
@@ -212,12 +136,137 @@ export default {
       emit(event, props.file);
     };
 
+    const { t } = useI18n();
+
+    const menuItems = [];
+
+    if (isViewable(props.file)) {
+      menuItems.push({
+        key: 1,
+        get text() {
+          return t("FileActionsCell.openViewerButtonText");
+        },
+        color: "var(--color-primary)"
+      });
+    }
+
+    if (!isFolder(props.file) && isConvertible(props.file)) {
+      if (!isModel(props.file)) {
+        menuItems.push({
+          key: 2,
+          get text() {
+            return t("FileActionsCell.createModelButtonText");
+          },
+          action: () => onClick("create-model")
+        });
+      } else {
+        menuItems.push({
+          key: 3,
+          get text() {
+            return t("FileActionsCell.removeModelButtonText");
+          },
+          action: () => onClick("remove-model")
+        });
+      }
+    }
+
+    menuItems.push({
+      key: 4,
+      get text() {
+        return t("FileActionsCell.renameButtonText");
+      },
+      action: () => onClick("update"),
+      disabled:
+        !props.project.isAdmin &&
+        props.file.user_permission < FILE_PERMISSION.READ_WRITE
+    });
+
+    menuItems.push({
+      key: 5,
+      get text() {
+        return t("FileActionsCell.downloadButtonText");
+      },
+      action: () => onClick("download"),
+      disabled:
+        !props.project.isAdmin &&
+        props.file.user_permission < FILE_PERMISSION.READ_WRITE
+    });
+
+    if (isFolder(props.file) && props.project.isAdmin) {
+      menuItems.push({
+        key: 6,
+        get text() {
+          return t("FileActionsCell.manageAccessButtonText");
+        },
+        action: () => onClick("manage-access")
+      });
+    }
+
+    if (
+      !isFolder(props.file) &&
+      (props.project.isAdmin ||
+        props.file.user_permission === FILE_PERMISSION.READ_WRITE)
+    ) {
+      menuItems.push({
+        key: 7,
+        get text() {
+          return t("FileActionsCell.VisaButtonText");
+        },
+        action: () => onClick("open-visa-manager")
+      });
+    }
+
+    if (
+      !isFolder(props.file) &&
+      (props.project.isAdmin ||
+        props.file.user_permission === FILE_PERMISSION.READ_WRITE)
+    ) {
+      menuItems.push({
+        key: 8,
+        get text() {
+          return t("FileActionsCell.addTagsButtonText");
+        },
+        action: () => onClick("open-tag-manager"),
+        dataTestId: "btn-open-tag-manager"
+      });
+    }
+
+    if (
```

Comment:
This `if` statement refers to the exact same condition as the previous one.
I think we could merge them into one.

Reply:
**Note:** you can also use the [`hasAdminPerm` helper](https://github.com/bimdata/platform-next/blob/develop/src/utils/file-structure.js#L288) to simplify the code here. ;)